test/calibration/test_calibrations.py

```diff
@@ -894,6 +909,33 @@ def test_single_control_channel(self):
 
         self.assertEqual(self.cals.get_schedule("tcp", (3, 2)), expected)
 
+    def test_inst_map_stays_consistent(self):
```

Comment:
I think it is a bug that this test is passing currently. The only reason it is passing is that the sigma parameters between the cr and xp schedules are tied, which is not realistic. If you uncouple the sigma parameters or switch to updating `amp` instead of sigma, the test fails.

When a parameter in a schedule nested inside a call instruction of an outer schedule is updated, the outer schedule also needs to be updated in the instruction schedule map. Should we address that case in this PR as well?

It also feels a bit weird that updating a parameter for a one qubit schedule also updates the parameter for two qubits. I thought we had scoped parameters to a number of qubits but perhaps not.

Reply:
Thanks for catching this. We should fix this in this PR. Here is how the test should look like: [eace06e](https://github.com/Qiskit/qiskit-experiments/pull/740/commits/eace06eeff7c9d09e923b77274c6b41369fe32ea)

Reply:
Ok this was non-trivial to fix. See here: [f3b76b9](https://github.com/Qiskit/qiskit-experiments/pull/740/commits/f3b76b9982f22e9079287de674149fa1068721dc)

The complications were:

1. figuring out which schedules call a schedule that we want to update and then add the caller to the update list.
2. figuring out the qubits of the caller schedules. This happens in e.g. with an ECR gate that calls an X gate. Suppose we want to update `xp` on qubit `(3, )`. Now suppose that `cr` on `(3, 2)` calls `xp` then under step 1. we find that we need to also update `cr` but we only know that we have qubits `(3, )` we therefore need to figure out that `cr` applies, e.g., on `(2, 3)` and `(3, 2)`. Out of safety both are updated here.